docs/pages/docs/usage/messages.mdx

```diff
@@ -278,3 +278,62 @@ Messages are always parsed and therefore e.g. for rich text you need to supply t
 </Callout>
 
 The value of a raw message can be any valid JSON value: strings, booleans, objects and arrays.
+
+## Right-to-left languages
+
+Languages such as Arabic, Hebrew and Persian use [right-to-left script](https://en.wikipedia.org/wiki/Right-to-left_script) (often abbreviated as RTL). For these languages, writing begins on the right side of the page and continues to the left.
+
+In addition to providing translated messages, proper RTL localization requires:
+
+1. Providing [the `dir` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/dir) on the document
+2. Support layout mirroring, e.g. by using [CSS logical properties](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_logical_properties_and_values)
+3. Support element mirroring, e.g. by customizing icons
+
+To handle these cases in your components, you can set up a reusable hook:
+
+```tsx filename="useTextDirection.tsx"
+import detectRtl from 'rtl-detect';
+import {useLocale} from 'next-intl';
+
+export default function useTextDirection(locale) {
+  const defaultLocale = useLocale();
+  if (!locale) locale = defaultLocale;
+  return detectRtl(locale) ? 'rtl' : 'ltr';
+}
+```
+
+… and use it accordingly:
+
+```tsx filename="app/[locale]/layout.tsx"
+import useTextDirection from './useTextDirection';
+
```

Comment:
Since the file extension of `app/[locale]/layout.tsx` is `.tsx` we need to type the params:

```tsx
interface LayoutProps {
  children: React.ReactNode;
  params: {
    locale: string;
  }
}

export default function Layout({children, params: {locale}}: LayoutProps) {
// ...
```

Reply:
`next-intl` currently doesn't take a stand in the docs on how props should be typed (I've seen many different preferences of developers). I think the Next.js docs do a good job of explaining this.